sessionctx/binloginfo/binloginfo_test.go

```diff
@@ -561,7 +561,7 @@ func (s *testBinlogSuite) TestAddSpecialComment(c *C) {
 }
 
 func mustGetDDLBinlog(s *testBinlogSuite, ddlQuery string, c *C) (matched bool) {
-	for i := 0; i < 10; i++ {
+	for i := 0; i < 15; i++ {
```

Comment:
Retry with sleep is bad, because it increases execution time and brings uncertainty to the unit test.
Can you use failpoint to hack it?

Reply:
Writing binlog is asynchronous in 2pc phase, add a `failpoint` in here will be complicated.

Reply:
Writing binlog is asynchronous in 2pc phase, add a `failpoint` in here will be complicated.